app/src/main/java/com/nutomic/syncthingandroid/service/SyncthingService.java

```diff
@@ -578,6 +595,30 @@ private void onServiceStateChange(State newState) {
         });
     }
 
+    public void registerOnRunConditionCheckResultChange(OnRunConditionCheckResultListener listener) {
+        listener.onRunConditionCheckResultChanged(mCurrentCheckResult.get());
+        mOnRunConditionCheckResultListeners.add(listener);
+    }
+
+    public void unregisterOnRunConditionCheckResultChange(OnRunConditionCheckResultListener listener) {
+        mOnRunConditionCheckResultListeners.remove(listener);
+    }
+
+    private void onRunConditionCheckResultChange(RunConditionCheckResult result) {
+        mHandler.post(() -> {
+            for (Iterator<OnRunConditionCheckResultListener> i = mOnRunConditionCheckResultListeners.iterator();
+                 i.hasNext(); ) {
+                OnRunConditionCheckResultListener listener = i.next();
+                if (listener != null) {
+                    listener.onRunConditionCheckResultChanged(result);
+                } else {
+                    i.remove();
```

Comment:
Why remove here?

Reply:
To be honest, I just copied the existing method and did not think too much about it.

Reply:
If an activity that registered a listener is unavailable being set to null, it's automatically cleant up. Nutomic's code has this all over the place.

Reply:
I've not addressed this one, since it's consistent with the other code. If it really is a blocker, I can of course adjust.

Reply:
I understand the value gets gc'ed and becomes null, but how given there is a pointer in this set, which should prevent collection?

If we had a weakref, I'd understand it, but we don't have a weakref, we have a strong ref.